readthedocs/projects/models.py

```diff
@@ -167,6 +168,20 @@ class AddonsConfig(TimeStampedModel):
         help_text="Enable/Disable all the addons on this project",
     )
 
+    options_doctool_name = models.CharField(
+        choices=DOCTOOL_NAME_CHOICES,
+        null=True,
+        blank=True,
+        max_length=128,
+        help_text="Name of the documentation tool used by this project",
+    )
+    options_doctool_root_selector = models.CharField(
```

Comment:
This is a UX issue, but it feels weird to expose these at the top level of Addons. I think most users don't have any idea what to do with this, especially since the UI has help_text hidden, and there's no HTML placeholder. If we're going to be surfacing this so prominently, we definitely need more UI to guide users about what to put here, but I'm pretty skeptical that most users will need to adjust this, and will know when and how to do it?

Reply:
Agreed this is an odd top level configuration. I felt the same and didn't mention this, but I'd probably put this under an "Advanced" tab. Neither field seems very approachable.

The doc tool field still feels odd to me as I think about it more though. It already feels redundant to the project configuration file, where most projects already define the doc tool.

Maybe a bigger point though, Sphinx is unique in that the HTML output is standardized across themes. Other doctools _depend on themes for HTML structure_, so we can't just ask for the doc tool to infer the CSS selector. _We need to ask doc tool + theme, which seems rather unmaintainable_.

What we really need out of this is just for themes to be able to dictate the main content. This would either be with the theme using an HTML structure that matches our generic detection, or by specifying the selector we use.

So it's feeling less like we should ask the user for the tool at all. In fact, this seems like the CSS selector field should be a field with two options:

- "Detect the main content automatically"
- "Use a custom value"

The variance in HTML layouts is not so great that we couldn't detect the content area with heuristics. This would be an overall better UX for most projects already, but really this feels like it should come from the theme ultimately, either way. I don't see a doctool field being used in this scenario.

Reply:
Yea, I think getting smarter about how we're detecting content is definitely a better use of our time than trying to get users to understand and use these fields. 

> Agreed this is an odd top level configuration. I felt the same and didn't mention this, but I'd probably put this under an "Advanced" tab. Neither field seems very approachable.

Yea, I think that makes more sense than the proposed UI :+1:

> So it's feeling less like we should ask the user for the tool at all. In fact, this seems like the CSS selector field should be a field with two options:

> "Detect the main content automatically"
> "Use a custom value"

Strong agree here. 💯

Reply:
I agree with everything you said here.

My thought behind asking for the doctool was because I think the user is more likely to know the doctool they are using than the CSS selector for the main content. Knowing the doctool gives us more data about what could be the correct CSS selector.

I wanted to avoid building a "magic main content detector" because it could be harder to maintain and keep consistent over time; but it probably has the same issue than maintaining the default value for each of the doctools 😄

Reply:
We could use the code we are using for EthicalAds to detect the theme (https://github.com/readthedocs/addons/blob/30392906ea9b75b5f4b58fd360d7964f81ba0f90/src/ethicalads.js#L41-L110) to auto-detect the doctool and theme, and build the CSS selector based on that when the value is "Detect the main content automatically".